doc/fwutil/fwutil.md

```diff
@@ -462,6 +516,218 @@ Aborted!
 
 **Note:** the default option is _--image=current_
 
+#### 2.2.2.4 Auto-update commands
+
+##### 2.2.2.4.1 Overview
+The purpose of the auto-update commands group is to provide and interface for automatic fw updates of various platform components based on the boot option and the platform firmwareupdate configuration file - platform_components.json.
+
+The existing FWutil infra supports the upgrade of platform components which can be performed during run time.
+The SSD upgrade for certain platforms requires the unmount of the filesystems and this needs to be incorporated as part of the reboot scripts.
+To integrate the upgrade of these platform components which can’t be performed during run time,
+the FWutil requires few more enhancements that specifies the reboot-type during which the user expects the upgrade to happen.
+The FWutil infra section below indicates the necessary changes to add support for auto update during reboot.
+The Platform plugins section describes the plugins that a platform must implement to support auto updates when the platform component APIs are not available.
+
+Automatic FW installation requires default platform_components.json to be created and placed at:
+_sonic-buildimage/device/<platform_name>/<onie_platform>/platform_components.json_
+default image path = /usr/share/sonic/device/<onie_platform>/
+
+Auto-update command can support the standalone custom firmware image package with --fw-image option.
+The package can be any format between `.tar` or `.tar.gz` and should have the `platform_components.json`, the firmware image(s).
+The package can have the auto-update plugin if the platform doesn't support platform api or if plugin needs to be updatedplugin.
+The package can also have the component upgrade utility(script) and the utility will be used for the component firmware upgrade if it's specified in `platform_compnents.json`.
+The `fwutil auto-update` commands uncompress the package and parse the `platform_components.json` to retrieve the firmware information.
+
+If the script path is available for the component firmware configuration in the `platform_components.json`,
+it means that the specific component firmware upgrade shall use the script to process the fwutil commands.
+`2.2.2.4.3 Platform Component Firmware Update Utility` explains the requirement of the component firmware upgrade utility
+to interfere with the fwutil to support the auto-update command - mainly status and install.
+
+**Note:**
+1. FW update will be disabled if component definition is not provided (e.g., 'BIOS': { })
+2. FW version will be read from image if `version` field is not provided
+
+##### 2.2.2.4.2 Description
+
+**The following command updates FW Automatically with installing the available component firmware, creating task(s) for update during reboot:**
+1. auto-update command for installation and reboot task creation
+```bash
+root@sonic:~# fwutil auto-update fw --yes --image=<current|next> --boot=<boot_type>
```

Comment:
@sujinmkang what if this will be called multiple times with different reboot types? Do we expect previously scheduled task will be cancelled?

Reply:
@nazariig no we're not going to allow another auto-update if the previous auto update was successful since the previous auto-update could have dependency with boot-type. As an example, the previous auto-update was called for cold reboot type which could've installed the cpld firmware but if a new auto-update request is initiated with fast-reboot, we don't want to proceed. So fwutil check if any task file is available, then a new auto-update request will not be allowed.